src/database/DataTypes/RequisitionItem.js

```diff
@@ -230,6 +230,135 @@ export class RequisitionItem extends Realm.Object {
   get lastRequisitionDate() {
     return this.item?.lastRequisitionDate;
   }
+
+  resetClosingVariables() {
+    this.openingStock = 0;
+    this.outgoingStock = 0;
+    this.incomingStock = 0;
+    this.positiveAdjustments = 0;
+    this.negativeAdjustments = 0;
+  }
+
+  setIncomingStock(value) {
+    const newClosing =
+      this.openingStock +
+      value -
+      this.outgoingStock +
+      this.positiveAdjustments -
+      this.negativeAdjustments;
+
+    const canUpdate = newClosing >= 0;
+
+    if (canUpdate) {
+      this.incomingStock = value;
+    } else {
+      this.resetClosingVariables();
+    }
+
+    return canUpdate;
```

Comment:
Trying to see if I can get away with just resetting all these values to 0 when theres a problem, rather than playing around with logic to find some values to put in the cells..